Core/src/Geometry/TrackingVolume.cpp

```diff
@@ -484,6 +484,27 @@ Acts::TrackingVolume::compatibleBoundaries(
       return BoundaryIntersection();
     }
 
+    if (options.forceIntersectBoundaries and
+        sIntersection.intersection.pathLength * options.navDir > 0) {
```

Comment:
does this do the same as `std::abs(sIntersection.intersection.pathLength) >0`? in that case we could also check for `sIntersection.intersection.pathLength != 0` if you think that is preferable

Reply:
This is to check that the boundary surface you want to intersect is along the direction, hence `sIntersection.intersection.pathLength` is required to have the same sign of `options.navDir`.